dml_events.go

```diff
@@ -406,17 +412,37 @@ func Int64Value(value interface{}) (int64, bool) {
 //
 // ref: https://github.com/mysql/mysql-server/blob/mysql-5.7.5/mysys/charset.c#L963-L1038
 // ref: https://github.com/go-sql-driver/mysql/blob/9181e3a86a19bacd63e68d43ae8b7b36320d8092/utils.go#L717-L758
-func appendEscapedString(buffer []byte, value string) []byte {
+//
+// We need to support right-padding of the generated string using 0-bytes to
+// mimic what a MySQL server would do for BINARY columns (with fixed length).
+//
+// ref: https://github.com/Shopify/ghostferry/pull/159
+//
+// This is specifically mentioned in the the below link:
+//
+//    When BINARY values are stored, they are right-padded with the pad value
+//    to the specified length. The pad value is 0x00 (the zero byte). Values
+//    are right-padded with 0x00 for inserts, and no trailing bytes are removed
+//    for retrievals.
+//
+// ref: https://dev.mysql.com/doc/refman/5.7/en/binary-varbinary.html
+func appendEscapedString(buffer []byte, value string, rightPadLengthForBinaryColumn uint) []byte {
```

Comment:
I think moving the 0-byte padding into a separate function would be cleaner, as the padding doesn't really have anything to do with properly escaping strings. That way, we only need to call it `if column.Type == schema.TYPE_BINARY` and don't need to modify the case of `decimal.Decimal` 

Reusing the naming, we can make a function like:

```go
 func rightPadLengthForBinaryColumn(buffer []byte, columnLength uint) []byte {
    ...
}
```

and just move the logic there.  It can be called directly after `appendEscapedString` in line 357.

Reply:
true - seems cleaner and avoids the other invocation of the method with the awkward `0` parameter

Reply:
actually, let me take that back. I agree it's cleaner, but the method currently does the quoting (as well as escaping the `'` ) and knows  about the length of the string

sure, all of that can be broken apart, but does that make it more readable?

Reply:
You're right. I think handling the trailing `'\''` that is appended to the `buffer` in a separate method would make this less readable.

Another option would be to pass the `column` to the `appendEscapedString` method and conditionally handle the padding in that method:

```go
// We need to support right-padding of the generated string using 0-bytes to
// mimic what a MySQL server would do for BINARY columns (with fixed length).
//
// ref: https://github.com/Shopify/ghostferry/pull/159
//
// This is specifically mentioned in the the below link:
//
//    When BINARY values are stored, they are right-padded with the pad value
//    to the specified length. The pad value is 0x00 (the zero byte). Values
//    are right-padded with 0x00 for inserts, and no trailing bytes are removed
//    for retrievals.
//
// ref: https://dev.mysql.com/doc/refman/5.7/en/binary-varbinary.html
func rightPadLengthForBinaryColumn(buffer []byte, padLength int) []byte {
	// continue 0-padding up to the desired length as provided by the
	// caller
	for i := 0; i < int(padLength); i++ {
		buffer = append(buffer, '\x00')
	}

	return buffer
}

// appendEscapedString replaces single quotes with quote-escaped single quotes.
// When the NO_BACKSLASH_ESCAPES mode is on, this is the extent of escaping
// necessary for strings.
//
// ref: https://github.com/mysql/mysql-server/blob/mysql-5.7.5/mysys/charset.c#L963-L1038
// ref: https://github.com/go-sql-driver/mysql/blob/9181e3a86a19bacd63e68d43ae8b7b36320d8092/utils.go#L717-L758
func appendEscapedString(buffer []byte, value string, column schema.TableColumn) []byte {
	buffer = append(buffer, '\'')

	var i int
	for i = 0; i < len(value); i++ {
		c := value[i]
		if c == '\'' {
			buffer = append(buffer, '\'', '\'')
		} else {
			buffer = append(buffer, c)
		}
	}

	if column.Type == schema.TYPE_BINARY {
		padLength := int(column.FixedSize) - i
		buffer = rightPadLengthForBinaryColumn(buffer, padLength)
	}

	return append(buffer, '\'')
}

```

Reply:
IMHO `appendEscapedString` is a pure string/conversion method and should not have to worry about columns or types. I have refactored out the trivial method for 0-padding now, but I feel making the former method aware of columns or types seems going the wrong direction (making the method less useful in other scenarios)

it's a nit and I'm fine either way, but I feel passing the column is wrong here

Reply:
I don't feel strongly - just that it seemed a bit awkward to be forcing a param for all cases when not all applied. Either way can work 👍